src/recipient.js

```diff
@@ -55,6 +56,24 @@ function Recipient(name) {
   // years of earning history: total earnings / 35 / 12.
   this.monthlyIndexedEarnings = 0;
 
+  // The total credits earned, maximum of 40
+  this.earnedCredits = 0;
+
+  // The total of credits that would be earned in the planned work
+  this.plannedCredits = 0;
+
+  // total credits = earnedCredits + planned credits
+  this.totalCredits = 0;
+
+  // Have earnings before 1978
+  this.earningsBefore1978 = false;
+
+  // Additional years of work needed, -1 means not earning enough to get a credit
+  this.neededYears = -1;
+
+  // Last year of credits needed to get to 40
+  this.lastYearOfEarnedCredits = 0;
```

Comment:
I don't think this gets used anywhere. You set it, but never read it.

Reply:
It was in the UI at one point, but was removed.  Removed.